src/Sentry/Protocol/SentryLog.cs

```diff
@@ -0,0 +1,280 @@
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Internal;
+
+namespace Sentry.Protocol;
+
+/// <summary>
+/// Represents the Sentry Log protocol.
+/// <para>This API is experimental and it may change in the future.</para>
+/// </summary>
+[Experimental(DiagnosticId.ExperimentalFeature)]
+public sealed class SentryLog : ISentryJsonSerializable
+{
+    private readonly Dictionary<string, SentryAttribute> _attributes;
+    private readonly SentryLogLevel _level;
+
+    [SetsRequiredMembers]
+    internal SentryLog(DateTimeOffset timestamp, SentryId traceId, SentryLogLevel level, string message)
+    {
+        Timestamp = timestamp;
+        TraceId = traceId;
+        Level = level;
+        Message = message;
+        _attributes = new Dictionary<string, SentryAttribute>(7);
+    }
+
+    /// <summary>
+    /// The timestamp of the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <remarks>
+    /// Sent as seconds since the Unix epoch.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public required DateTimeOffset Timestamp { get; init; }
+
+    /// <summary>
+    /// The trace id of the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public required SentryId TraceId { get; init; }
+
+    /// <summary>
+    /// The severity level of the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public required SentryLogLevel Level
+    {
+        get => _level;
+        init
+        {
+            SentryLogLevelExtensions.ThrowIfOutOfRange(value);
```

Comment:
Ideally we don't throw from public APIs (and also properties ideally shouldnt' throw either, they 'look like fields').
Possibly better approach here is to debug log if diagnostic logging is on, and default to some level like `debug`.

Reply:
removed all related throw helpers and added tests to ensure fallback behavior when out of range